earth_enterprise/src/fusion/portableglobe/servers/fileunpacker/shared/glc_unpacker.cpp

```diff
@@ -333,6 +333,31 @@ void GlcUnpacker::MapDataPacketWalker(int layer, const map_packet_walker& walker
   iter->second->MapDataPacketWalker(layer, walker);
 }
 
+/**
+ * Find the names of all files in each layer of a globe.
+ */
+void GlcUnpacker::MapFileWalker(const map_file_walker& walker)
+{
+    for( auto& index_item : unpacker_index_) {
+        MapFileWalker(index_item.first, walker);
```

Comment:
you need a way to break out of this if the call returned false in the inner loop.  The packet walker handles this by having the layer version of the walker implemented in `FileUnpacker` and that method returns a boolean which signals if the callback indicated a stop in the inner loop.